### PR TITLE
Rework configuration responsibilities, group packages by type, hook up parser

### PR DESCRIPTION
**Pull request by allmarkedup**

### CODE REVIEW

File: packages/cli/cli/bin/fractal [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  const cli = new Cli({
+    configPath,
+    config: cliConfig
+  });
+
+  /*
+   * Load and register extensions.
+   */
+
+  const extensions = loadExtensions(omit(config, ['fractal', 'cli']));
```
Comment (dkhuntrods) [inline packages/cli/cli/bin/fractal]: Was this to be `['app', 'cli']`?

File: packages/cli/cli/src/loaders/config.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /*
+    * package.json exists, load it and look for a config file.
+    * Custom config file names can be specified using
+    * the `fractal.cli.config` property in the package.json
+    */
+
+    const pkg = require(pkgPath);
+    const pkgConfig = pkg.fractal;
+    const fileNames = toArray(get(pkgConfig || {}, 'cli.config', opts.config));
+    const filePaths = fileNames.map(p => join(process.cwd(), p.replace(/^\.\//, '')));
```
Comment (dkhuntrods) [inline packages/cli/cli/src/loaders/config.js]: Will that `p.replace` be cross-platform?

File: packages/cli/cli/src/loaders/fractal.test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+describe('CLI fractal loader', function () {
+  it('exports a function', function () {
+    expect(loader).to.be.a('function');
+  });
+
+  it('returns a Fractal instance', function () {
+    expect(loader()).to.be.an.instanceOf(fractal.Fractal);
+  });
+
+  // it('attempts to load the fractal version from the CWD first', function () {
```
Comment (dkhuntrods) [inline packages/cli/cli/src/loaders/fractal.test.js]: Is this a WIP? Worth marking as pending so we don't forget about it?

Comment (allmarkedup) [inline packages/cli/cli/src/loaders/fractal.test.js]: Actually I was unsure how to go about writing a test this... any ideas on the best way to do it?

**Pull request merged**